File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/TeamsPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import java.util.List;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record TeamsPacket(String teamName, String teamDisplayName, String teamPrefix, String teamSuffix,
        byte mode,
        byte friendlyFire, List<String> players) implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeString(os, teamName);
        os.writeByte(mode);
        writeString(os, teamDisplayName, teamPrefix, teamSuffix);
        os.writeByte(friendlyFire);
        os.writeShort(players.size());
        writeString(os, players);
    }

    @Override
    public byte getId() {
        return 0x3E;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/TeamsPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class TeamsPacket(
    val teamName: String, val teamDisplayName: String, val teamPrefix: String, val teamSuffix: String,
    val mode: Byte,
    val friendlyFire: Byte, val players: List<String>
) : Packet.Outgoing {
    override fun serialize(os: ByteBuf) {
        os.writeString(teamName)
        os.writeByte(mode.toInt())
        os.writeString(teamDisplayName, teamPrefix, teamSuffix)
        os.writeByte(friendlyFire.toInt())
        os.writeShort(players.size)
        os.writeString(players)
    }

    override val id: Byte
        get() = 0x3E
}
